org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/TryWithResourcesFilter.java

```diff
@@ -0,0 +1,426 @@
+		private final Map<String, String> owners = new HashMap<String, String>();
+		private final Map<String, LabelNode> labels = new HashMap<String, LabelNode>();
+
+		private AbstractInsnNode start;
+		private AbstractInsnNode cursor;
+
+		Matcher(final IFilterOutput output) {
+			this.output = output;
+		}
+
+		private enum JavacPattern {
+			OPTIMAL, FULL, OMITTED_NULL_CHECK, METHOD,
+		}
+
+		private void start(final AbstractInsnNode start) {
+			this.start = start;
+			cursor = start.getPrevious();
+			vars.clear();
+			labels.clear();
+			owners.clear();
+		}
+
+		private boolean matchJavac(final JavacPattern p) {
+			// "catch (Throwable t)"
+			if (!nextIsVar(Opcodes.ASTORE, "t1")) {
+				return false;
+			}
+			// "primaryExc = t"
+			if (!nextIsVar(Opcodes.ALOAD, "t1")) {
+				return false;
+			}
+			if (!nextIsVar(Opcodes.ASTORE, "primaryExc")) {
+				return false;
+			}
+			// "throw t"
+			if (!nextIsVar(Opcodes.ALOAD, "t1")) {
+				return false;
+			}
+			if (!nextIs(Opcodes.ATHROW)) {
+				return false;
+			}
+
+			// "catch (any t)"
+			if (!nextIsVar(Opcodes.ASTORE, "t2")) {
+				return false;
+			}
+			if (!nextIsJavacClose(p)) {
+				return false;
+			}
+			// "throw t"
+			if (!nextIsVar(Opcodes.ALOAD, "t2")) {
+				return false;
+			}
+			if (!nextIs(Opcodes.ATHROW)) {
+				return false;
+			}
+
+			final AbstractInsnNode end = cursor;
+			AbstractInsnNode c = start.getPrevious();
+			cursor = c;
+			while (!nextIsJavacClose(p)) {
+				c = c.getPrevious();
+				cursor = c;
+				if (cursor == null) {
+					return false;
+				}
+			}
+
+			AbstractInsnNode m = cursor;
+			next();
+			if (cursor.getOpcode() != Opcodes.GOTO) {
+				cursor = m;
+			}
+
+			output.ignore(c.getNext(), cursor);
+			output.ignore(start.getNext(), end);
+			return true;
+		}
+
+		/**
+		 * On a first invocation will associate variables with names "r" and
+		 * "primaryExc", on subsequent invocations will use those associations
+		 * for checks.
+		 */
+		private boolean nextIsJavacClose(final JavacPattern p) {
+			switch (p) {
+			case METHOD:
+			case FULL:
+				// "if (r != null)"
+				if (!(nextIsVar(Opcodes.ALOAD, "r")
+						&& nextIs(Opcodes.IFNULL))) {
+					return false;
+				}
+			}
+			switch (p) {
+			case METHOD:
+			case OPTIMAL:
+				if (nextIsVar(Opcodes.ALOAD, "primaryExc")
+						&& nextIsVar(Opcodes.ALOAD, "r")
+						&& nextIs(Opcodes.INVOKESTATIC)) {
+					final MethodInsnNode m = (MethodInsnNode) cursor;
+					return "$closeResource".equals(m.name)
+							&& "(Ljava/lang/Throwable;Ljava/lang/AutoCloseable;)V"
+									.equals(m.desc);
+				}
+				return false;
+			case FULL:
+			case OMITTED_NULL_CHECK:
+				return nextIsVar(Opcodes.ALOAD, "primaryExc")
+						// "if (primaryExc != null)"
+						&& nextIs(Opcodes.IFNULL)
+						// "r.close()"
+						&& nextIsClose("r") && nextIs(Opcodes.GOTO)
+						// "catch (Throwable t)"
+						&& nextIs(Opcodes.ASTORE)
```

Comment:
@marchof I assume that in https://github.com/jacoco/jacoco/pull/500#discussion_r108958800 you refer to this `ASTORE` and to its pair - following `ALOAD`. Second remaining `ALOAD` is in code for ecj - I'll check it separately.

Here it is probably a bit more tricky than for previous: method `nextIsJavacClose` is called to check bytecode of both an exceptional and non-exceptional paths, and I suppose those instructions might use different variables on different paths, so would have to have different names, but should be doable.